DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToHost/Raw/Geometry/ICurveConverterToHost.cs

```diff
@@ -0,0 +1,128 @@
+using Objects;
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+
+namespace Speckle.Converters.RevitShared.ToSpeckle;
+
+public class ICurveConverterToHost : ITypedConverter<ICurve, DB.CurveArray>
+{
+  private readonly ITypedConverter<SOG.Point, DB.XYZ> _pointConverter;
+  private readonly ITypedConverter<SOG.Vector, DB.XYZ> _vectorConverter;
+  private readonly ITypedConverter<SOG.Arc, DB.Arc> _arcConverter;
+  private readonly ITypedConverter<SOG.Line, DB.Line> _lineConverter;
+  private readonly ITypedConverter<SOG.Circle, DB.Arc> _circleConverter;
+  private readonly ITypedConverter<SOG.Ellipse, DB.Curve> _ellipseConverter;
+  private readonly ITypedConverter<SOG.Polyline, DB.CurveArray> _polylineConverter;
+  private readonly ITypedConverter<SOG.Curve, DB.Curve> _curveConverter;
+
+  public ICurveConverterToHost(
+    ITypedConverter<SOG.Point, DB.XYZ> pointConverter,
+    ITypedConverter<SOG.Vector, DB.XYZ> vectorConverter,
+    ITypedConverter<SOG.Arc, DB.Arc> arcConverter,
+    ITypedConverter<SOG.Line, DB.Line> lineConverter,
+    ITypedConverter<SOG.Circle, DB.Arc> circleConverter,
+    ITypedConverter<SOG.Ellipse, DB.Curve> ellipseConverter,
+    ITypedConverter<SOG.Polyline, DB.CurveArray> polylineConverter,
+    ITypedConverter<SOG.Curve, DB.Curve> curveConverter
+  )
+  {
+    _pointConverter = pointConverter;
+    _vectorConverter = vectorConverter;
+    _arcConverter = arcConverter;
+    _lineConverter = lineConverter;
+    _circleConverter = circleConverter;
+    _ellipseConverter = ellipseConverter;
+    _polylineConverter = polylineConverter;
+    _curveConverter = curveConverter;
+  }
+
+  public DB.CurveArray Convert(ICurve target)
+  {
+    DB.CurveArray curveArray = new();
+    switch (target)
+    {
+      case SOG.Line line:
+        curveArray.Append(_lineConverter.Convert(line));
+        return curveArray;
+
+      case SOG.Arc arc:
+        curveArray.Append(_arcConverter.Convert(arc));
+        return curveArray;
+
+      case SOG.Circle circle:
+        curveArray.Append(_circleConverter.Convert(circle));
+        return curveArray;
+
+      case SOG.Ellipse ellipse:
+        curveArray.Append(_ellipseConverter.Convert(ellipse));
+        return curveArray;
+
+      case SOG.Spiral spiral:
+        return _polylineConverter.Convert(spiral.displayValue);
+
+      case SOG.Curve nurbs:
+        var n = _curveConverter.Convert(nurbs);
+
+        // poc : in original converter, we were passing a bool into this method 'splitIfClosed'.
```

Comment:
IIRC @AlanRynne was explaining something similar to this before.. by saying smth like "there is no closed curve definition.. (maybe I'm wrong)"

Reply:
Revit does not support closed nurbs curves/splines. So the split logic needs to come back one way or another.

Reply:
If it's complex I am okay this being broken into a separate ticket (DUI3 project) and the link dropped here